src/MonoTorrent.Tests/MonoTorrent.Client.Modes/MetadataModeTests.cs

```diff
@@ -112,12 +113,36 @@ public async Task RequestMetadata ()
             Assert.AreEqual (rig.Torrent.InfoHash, new InfoHash (new SHA1Managed ().ComputeHash (stream)), "#1");
         }
 
+        [Test]
+        public async Task AfterHandshake_SendBitfieldMessage()
+        {
+            var torrent = Path.Combine (AppDomain.CurrentDomain.BaseDirectory, "file.torrent");
+            await Setup (true, torrent);
+            await SendMetadataCore (torrent, new BitfieldMessage (rig.Torrent.Pieces.Count));
```

Comment:
@vlasenkoalexey This should solve your problem. This means that monotorrent can retrieve metadata from leechers in all circumstances. Without this patch it could only download metadata from seeders or leechers which have not downloaded any data. The tests, unfortunately, didn't exercise that case.